src/cargo/core/resolver/encode.rs

```diff
@@ -56,8 +56,8 @@ impl EncodableResolve {
                     let to_depend_on = try!(edge.to_package_id(default));
                     let precise_pkgid =
                         tmp.get(&to_depend_on)
-                           .map(|p| to_depend_on.with_precise(p.clone()))
-                           .unwrap_or(to_depend_on.clone());
+                           .map_or(to_depend_on.clone(),
```

Comment:
Hm I personally find `.map(x).unwrap_or(y)` clearer to understand than `map_or` (because the order of arguments is "backwards" for the `map_or` case)...